File: app/src/main/java/com/fstyle/structure_android/data/source/remote/BaseRemoteDataSource.java
package com.fstyle.structure_android.data.source.remote;

import com.fstyle.structure_android.data.source.remote.api.service.NameApi;

/**
 * Created by le.quang.dao on 10/03/2017.
 */

public abstract class BaseRemoteDataSource {

    NameApi mNameApi;

    public BaseRemoteDataSource(NameApi nameApi) {
        mNameApi = nameApi;
    }
}

File: app/src/main/java/com/fstyle/structure_android/data/source/remote/BaseRemoteDataSource.kt
package com.fstyle.structure_android.data.source.remote

import com.fstyle.structure_android.data.source.remote.api.service.NameApi

/**
 * Created by le.quang.dao on 10/03/2017.
 */

abstract class BaseRemoteDataSource(internal var nameApi: NameApi) {
  fun getNameApi(): NameApi {
    return nameApi
  }
}
